java-maven/maven-perforce/README.md

```diff
@@ -0,0 +1,161 @@
+    <scm>
+        <connection>scm:p4:rookout.com:1666://depot/rookout/maven-example</connection>
+    </scm>
+    ```
+4. Export your Rookout Token
+     ```bash
+     export ROOKOUT_TOKEN=<Your-Token>
+     ```
+5. Downloading the Rookout Java Agent from available on [maven central]::
+    ```bash
+    $ wget "http://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.rookout&a=rook&v=LATEST"  -O rook.jar
+    ```  
+6. Specifying the Rookout Java Agent when running:
+    ```bash
+    $ java  -javaagent:rook.jar -jar target/rookoutDemo-1.0.0-jar-with-dependencies.jar
+    ```
+
+7. Make sure everything worked: [http://localhost:7000/](http://localhost:7000/hello)
+
+8. Go to [http://app.rookout.com](http://app.rookout.com) and start debugging! 
+
+## Rookout Integration explained
+
+This example is based of the Java javalin "Hello-World" example available [here].
+
+1. We added the source to the project jar(`pom.xml`) to enable rookout:
+    ```xml
+            <resources>
+                <resource>
+                    <directory>${basedir}/src/main/java</directory>
+                </resource>
+            </resources>
+ 
+   ```
+2. In order for Rookout's agent to sync with the right source code version, in this project we udpate the JAR's MANIFEST with the depot path and changelist:
+    ```xml
+    <Rookout-Revision>${buildNumber}</Rookout-Revision>
+    <Rookout-Repository>Perforce:${depotPath}</Rookout-Repository>
+    ```
+3. All of the MANIFEST update is accomplished using several plugins. We have added the following to the application's original pom.xml:
+    Required properties:    
+    ```xml
+     <properties>
+        <!-- p4maven version, for easy version upgrade -->
+        <p4maven.version>1.0.6</p4maven.version>
+        <!-- Your Perforce depot path -->
+        <depotPath>//depot/path/to/project</depotPath>
+    </properties>
+    ```
+    Source control connection string:
+    ```xml
+      <!-- Set your Perforce connection string -->
+    <scm>
+        <connection>scm:p4:[protocol:][[username[:password]@]hostname:port:]${depotPath}</connection>
+    </scm>
+    ```
+    Adding the following plugins into your plugins tag:
+    ```xml
+    <!-- setup the buildnumber plugin -->
+    <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>buildnumber-maven-plugin</artifactId>
+        <version>1.4</version>
+
+        <dependencies>
+            <dependency>
+                <groupId>com.perforce.p4maven</groupId>
+                <artifactId>p4maven-provider</artifactId>
+                <version>${p4maven.version}</version>
+            </dependency>
+        </dependencies>
+        <executions>
+            <execution>
+                <phase>validate</phase>
+                <goals>
+                    <goal>create</goal>
+                </goals>
+            </execution>
+        </executions>
+        <configuration>
+            <doCheck>false</doCheck>
```

Comment:
Why false? :)

Reply:
Do you want us to fail the customer's build process if he has modified files?
Feels intrusive imo, they are grownups, they can fail themselves...

Reply:
Can you add a comment explaining that it can be turned on?